### DISCUSSION TITLE
Simplified for loop

### DISCUSSION THREAD

**Discussion by wojciechsura** [dotnet/csharplang#7535] [Language Ideas]
I'm proposing a simplified for loop in cases that something needs to be done x times, but we don't care about the iterator.

Let's consider the following scenario:

```csharp
int count = binaryReader.ReadInt32();
for (int i = 0; i < count; i++)
{
    // Read i-th entry
}
```

The simplified notation would allow the following:

```csharp
int count = binaryReader.ReadInt32();
for (count)
{
    // Read entry
}
```

Rationale:
For loop is used if we care about _the index_ of processed item. Foreach loop is used if we care about _the processed item_. However, there is no option if we only care about certain number of executions.
This is a syntax simplification on the same level as e.g. simplifying using statements so that they don't require braces if their scope equals to enclosing scope.

Alternative:
Not too many. The only one that comes to my mind is:

```csharp
foreach (var _ in Enumerable.Range(0, count))
{

}
```

However, this one is even longer than a regular for loop.

**Comment by huoyaoyuan** [comment]
Covered in #2810.

The use case is concrete, we can discuss for the syntax later.

Moreover, if the iteration count is constant, the compiler (roslyn or JIT) can choose to unroll it.
